reactos/drivers/usb/usbhub_new/ioctl.c

```diff
@@ -815,27 +815,27 @@ USBH_IoctlGetNodeConnectionAttributes(IN PUSBHUB_FDO_EXTENSION HubExtension,
     Status = STATUS_INVALID_PARAMETER;
 
     NumPorts = HubExtension->HubDescriptor->bNumberOfPorts;
-    Port = 1;
 
-    if (NumPorts > 0)
+    if (NumPorts == 0)
     {
-        while (Port != ConnectionIndex)
-        {
-            ++PortData;
-            ++Port;
+        goto Exit;
+    }
 
-            if (Port > NumPorts)
-            {
-                goto Exit;
-            }
+    for (Port = 1; Port != ConnectionIndex; ++Port)
```

Comment:
This loop (and the ones like it) is unnecessary. It just increases one integer waiting for it to become equal to another.

```
if (ConnectionIndex == 0 ||
    ConnectionIndex > NumPorts)
{
    goto Exit;
}
PortData =&HubExtension->PortData[ConnectionIndex - 1];
```